prog/dftb+/lib_dftbplus/main.F90

```diff
@@ -3142,12 +3236,13 @@ subroutine getFillingsAndBandEnergies(eigvals, nElectrons, nSpinBlocks, tempElec
       ! Fixed value of the Fermi level for each spin channel
       do iS = 1, nSpinHams
         call electronFill(Eband(iS:iS), fillings(:,:,iS:iS), TS(iS:iS), E0(iS:iS), Ef(iS),&
-            & eigvals(:,:,iS:iS), tempElec, iDistribFn, kWeights)
+            & eigvals(:,:,iS:iS), tempElec, iDistribFn, kWeights, tNonAufbau, &
+            & tSpinPurify, iDet, nEl, iS)
       end do
     else if (nSpinHams == 2 .and. tSpinSharedEf) then
       ! Common Fermi level across two colinear spin channels
       call Efilling(Eband, Ef(1), TS, E0, fillings, eigvals, sum(nElecFill), tempElec, kWeights,&
-          & iDistribFn)
+          & iDistribFn, tNonAufbau, tSpinPurify, iDet, nEl, iS)
```

Comment:
iS is not set in this call

Reply:
iS is needed by us in electronFill, which is called by this subroutine I thought